Feature/#86 實作個人看板的歷史留言

Amobe opened the pull request:
## 👏 解決掉的 issue / Resolved Issues
- close #86 

## 📝 相關的 issue / Related Issues
- #86 

## ⛏ 變更內容 / Details of Changes
- 新增 user_comment_record.go
  - NewUserCommentRecord 可判斷傳入的資料是否符合留言的格式，並從符合格式的資料中擷取留言者與時間。
- 新增 ansi.go
  - 小工具可以將 ANSI escape code 移除。
- 修改 bbs.go
  - 新增 UserCommentConnector interface 處理個人留言的快取(未實作細節)。
  - 新增 GetUserCommentRecordFile method 從所有開板中找出使用者的歷史留言。
  - 新增 GetBoardUserCommentRecord method 從特定看板中找出使用者的歷史留言。
  - 新增 GetBoardArticleCommentRecords method 從特定文章中找出所有使用者的留言。
  - 新增 bbs_test.go 利用 fakeConnector 模擬實際存取檔案的功能，藉此完成 DB 邏輯的測試。

Commit:
Implement personal comment history

Add user_comment_record.go to parse the owner and time from raw data.
Add ansi.go to filter the ANSI escape code for input data.

After removing the ANSI escape code, the owner and time can be parsed
from the data easily.

Implement methods to get user's comments from all boards and from the
specific board.
Implement a method to get all comments from a specific article.

PichuChen commented on ansi.go:
```diff
@@ -0,0 +1,26 @@
+
+var re = regexp.MustCompile(ansi)
+
+func FilterANSI(dst, src []byte) []byte {
```
這邊幾個函示的用途是？

Amobe replied:
這是忘記移除的 function。
原本認為處理 bytes 比較直覺，後來改為 string，效能比較好，變數也不會 escape to heap。

PichuChen commented on ansi_test.go:
```diff
@@ -0,0 +1,41 @@
+)
+
+func TestFilterANSI(t *testing.T) {
+	src := []byte("[1;31m→ [33mpichu2[m[33m:推")
```
如果是不可見字元，可以用 `\x1b` 之類的來表示會比較好？

Amobe replied:
的確比較好， `\x1b` 不是空白很容易搞混。

Commit:
Remove unuse ANSI bytes filter function

The function is replaced by FilterStringANSI function

Commit:
Replace invisible char to byte code

The invisible char is not space.
Replace it with bytecode to notice the difference.

PichuChen commented on ansi.go:
```diff
@@ -0,0 +1,18 @@
+	return re.ReplaceAllString(src, "")
+}
+
+func ToString(b []byte) string {
```
這邊不建議這樣做，容易出現很難抓的Bug

Amobe replied:
ok

Commit:
Remove unsafe byte string convertor

PichuChen reviewed (approved):
LGTM
